components/encryption/src/manager/mod.rs

```diff
@@ -98,75 +115,78 @@ impl Dicts {
         }
     }
 
-    fn save_key_dict(&mut self, master_key: &dyn Backend) -> Result<()> {
+    fn save_key_dict(&self, master_key: &dyn Backend, key_dict: &mut KeyDictionary) -> Result<()> {
         let file = EncryptedFile::new(&self.base, KEY_DICT_NAME);
         if !master_key.is_secure() {
-            for value in self.key_dict.keys.values_mut() {
+            for value in key_dict.keys.values_mut() {
                 value.was_exposed = true
             }
         }
-        let key_bytes = self.key_dict.write_to_bytes()?;
+        let key_bytes = key_dict.write_to_bytes()?;
         file.write(&key_bytes, master_key)?;
 
         ENCRYPTION_FILE_SIZE_GAUGE
             .with_label_values(&["key_dictionary"])
             .set(key_bytes.len() as _);
-        ENCRYPTION_DATA_KEY_GAUGE.set(self.key_dict.keys.len() as _);
+        ENCRYPTION_DATA_KEY_GAUGE.set(key_dict.keys.len() as _);
 
         Ok(())
     }
 
-    fn save_file_dict(&self) -> Result<()> {
+    fn save_file_dict(&self, file_dict: &mut FileDictionary) -> Result<()> {
         let file = EncryptedFile::new(&self.base, FILE_DICT_NAME);
-        let file_bytes = self.file_dict.write_to_bytes()?;
+        let file_bytes = file_dict.write_to_bytes()?;
         // File dict is saved in plaintext.
         file.write(&file_bytes, &PlaintextBackend::default())?;
 
         ENCRYPTION_FILE_SIZE_GAUGE
             .with_label_values(&["file_dictionary"])
             .set(file_bytes.len() as _);
-        ENCRYPTION_FILE_NUM_GAUGE.set(self.file_dict.files.len() as _);
+        ENCRYPTION_FILE_NUM_GAUGE.set(file_dict.files.len() as _);
 
         Ok(())
     }
 
-    fn get_key(&self, key_id: u64) -> Option<&DataKey> {
```

Comment:
shall we keep `get_key` and `current_data_key` methods here with the mutex locking logic?